src/legacy/core_plugins/console/np_ready/public/lib/ace_token_provider/token_provider.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IEditSession, TokenInfo } from 'brace';
+import { TokensProvider, Token, Position } from '../../interfaces';
+
+interface ActualTokenInfo extends TokenInfo {
+  type: string;
+}
+
+const toToken = (lineNumber: number, tokens: ActualTokenInfo[]): Token[] => {
+  let acc = '';
+  return tokens.map(token => {
+    const column = acc.length + 1;
+    acc += token.value || '';
```

Comment:
When is `token.value` falsey? It seems like we would get into a strange situation if this occurred, since appending an empty string won't change the length of `acc` and yet we're using the length of `acc` to specify `column` on line 30. So we'd have multiple tokens occupying the same column.

If `token.value` can never be falsey then maybe we should just remove this default value? Or, if it can be falsey, then I wonder if this represents an error state -- in which case maybe we should `throw`? Or if it's not an error state, should we just skip the token without mutating `acc` and without creating a token, and move on to the next one?

Reply:
Great catch! You are totally correct that this should be an error case. In fact, most tokenizers I've used will error out if you match an empty string or if it doesn't have a way to tokenize a string. Will remove the default.

Reply:
👍 Maybe some unit tests for this function or testing this error case through its consumers would be good too.